ddtrace/internal/writer.py

```diff
@@ -706,3 +708,66 @@ def _get_finalized_headers(self, count):
         headers = self._headers.copy()
         headers["X-Datadog-Trace-Count"] = str(count)
         return headers
+
+
+class CIVisibilityWriter(HTTPWriter):
+    RETRY_ATTEMPTS = 5
+    HTTP_METHOD = "PUT"
+    STATSD_NAMESPACE = "civisibilitywriter"
+
+    def __init__(
+        self,
+        intake_url,  # type: str
+        sampler=None,  # type: Optional[BaseSampler]
+        priority_sampler=None,  # type: Optional[BasePrioritySampler]
+        processing_interval=get_writer_interval_seconds(),  # type: float
+        timeout=agent.get_trace_agent_timeout(),  # type: float
+        dogstatsd=None,  # type: Optional[DogStatsd]
+        sync_mode=True,  # type: bool
+        report_metrics=False,  # type: bool
+        api_version=None,  # type: Optional[str]
+        reuse_connections=None,  # type: Optional[bool]
+        headers=None,  # type: Optional[Dict[str, str]]
+    ):
+        encoder = CIVisibilityEncoderV01(0, 0)
+        encoder.set_metadata(
+            {
+                "language": "python",
+                "env": config.env,
+                "runtime-id": get_runtime_id(),
+                "library_version": ddtrace.__version__,
+            }
+        )
+
+        headers = headers or dict()
+        headers["dd-api-key"] = os.environ.get("DD_API_KEY", "dummyapikey")
+
+        super(CIVisibilityWriter, self).__init__(
+            intake_url=intake_url,
+            endpoint="v0.5/traces",  # XXX
+            encoder=encoder,
+            sampler=sampler,
+            priority_sampler=priority_sampler,
+            processing_interval=processing_interval,
+            timeout=timeout,
+            dogstatsd=dogstatsd,
+            sync_mode=sync_mode,
+            reuse_connections=reuse_connections,
+            headers=headers,
+        )
+
+    def stop(self):
```

Comment:
Is there a reason to implement this? I don't see the `HTTPWriter` class implement this either and this ultimately inherits from `PeriodicService`, which implements its own `stop()`.

Reply:
It's implemented because `CIVisibilityWriter` defaults to synchronous mode, which doesn't `start()` the service. The conditional in this function avoids trying to stop a service that was never started.